client/src/app/components/task-manager/TaskManagerDrawer.tsx

```diff
@@ -0,0 +1,256 @@
+TaskManagerDrawer.displayName = "TaskManagerDrawer";
+
+const TaskStateToIcon: React.FC<{ task: TaskManagerTask }> = ({ task }) =>
+  task.state === "Ready" ? (
+    <Tooltip content="Ready">
+      <CheckCircleIcon />
+    </Tooltip>
+  ) : task.state === "Postponed" ? (
+    <Tooltip content="Postponed">
+      <PauseCircleIcon />
+    </Tooltip>
+  ) : task.state === "Pending" ? (
+    <Tooltip content="Pending">
+      <PendingIcon />
+    </Tooltip>
+  ) : task.state === "Running" ? (
+    <Tooltip content="Running">
+      <InProgressIcon />
+    </Tooltip>
+  ) : (
+    <Tooltip content="Unknown">
+      <TaskIcon />
+    </Tooltip>
+  );
+
+const TaskItem: React.FC<{
+  task: TaskManagerTask;
+  expanded: boolean;
+  onExpandToggle: (expand: boolean) => void;
+}> = ({ task, expanded, onExpandToggle }) => {
+  const starttime = dayjs(task.started ?? task.createTime);
+  const title = expanded
+    ? `${task.id} (${task.addon})`
+    : `${task.id} (${task.addon}) - ${task.applicationName} - ${
+        task.priority ?? 0
+      }`;
+
+  return (
+    <NotificationDrawerListItem
+      key={task.id}
+      variant="info"
+      className={css(
+        expanded && "task-manager-item-expanded",
+        !expanded && "task-manager-item-collapsed"
+      )}
+      onClick={() => onExpandToggle(!expanded)}
+    >
+      <NotificationDrawerListItemHeader
+        variant="custom"
+        title={title}
+        icon={<TaskStateToIcon task={task} />}
+      >
+        {/* Put the item's action menu here */}
+      </NotificationDrawerListItemHeader>
+      {expanded ? (
+        <NotificationDrawerListItemBody
+          timestamp={
+            <Tooltip content={starttime.format("ll, LTS")}>
+              <span>{starttime.fromNow()}</span>
+            </Tooltip>
+          }
+        >
+          <div>{task.applicationName}</div>
+          {/* TODO: Link to /applications with filter applied? */}
+          <div>Priority {task.priority}</div>
+          {/* TODO: Bucket to Low, Medium, High? */}
+        </NotificationDrawerListItemBody>
+      ) : undefined}
+    </NotificationDrawerListItem>
+  );
+};
+
+const useTaskManagerData = () => {
+  const [pageSize, setPageSize] = useState(PAGE_SIZE);
+  const increasePageSize = () => {
+    setPageSize(pageSize + PAGE_SIZE);
+  };
+
+  const { result, isFetching } = useServerTasks(
+    {
+      filters: [{ field: "state", operator: "=", value: "queued" }],
+      sort: {
+        field: "id",
+        direction: "desc",
+      },
+      page: {
+        pageNumber: 1,
+        itemsPerPage: pageSize,
+      },
+    },
+    5000
+  );
+
+  const tasks: TaskManagerTask[] = useMemo(
+    () =>
+      !result.data
+        ? []
+        : result.data.map(
+            (task) =>
+              ({
+                id: task.id ?? -1,
+                createUser: task.createUser ?? "",
+                updateUser: task.updateUser ?? "",
+                createTime: task.createTime ?? "", // TODO: date?
```

Comment:
1. with "flat" object approach it makes sense to pre-format the props - so I would use string-formatted dates
2. similar logic for retrieving/formatting values is used in [toCells mapper](https://github.com/rszwajko/tackle2-ui/blob/3ed1dd44038319e3e4742b8d0234b32a83cdd603/client/src/app/pages/tasks/tasks-page.tsx#L204) - we might won't to use common step at some point.

Reply:
Potential future work to harmonize the display formatting of the task's fields.

Reply:
I opened https://github.com/konveyor/tackle2-ui/issues/1969 to capture this work once all the main task functions are merged.